### PR TITLE
Recepie for a new Version of Scriptureguide

### PR DESCRIPTION
**Pull request by Paradoxianer**

### CODE REVIEW

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 1]
```diff
@@ -0,0 +1,60 @@
+SUMMARY="Bible study tool that supports a wide variety of Bibles"
+DESCRIPTION="ScriptureGuide is a Bible study tool. It supports all of the \
+features you'd expect: search, setting the font, taking notes, etc. Through \
+the accompanying ScriptureGuideManager app, you can download hundereds of free \
+copies of various Bibles, commentaries, and related books."
+HOMEPAGE="https://github.com/HaikuArchives/ScriptureGuide"
+COPYRIGHT="2019 ScriptureGuide Team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/Paradoxianer/ScriptureGuide/archive/v0.9.1.tar.gz"
```
Comment (korli) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]:
```suggestion
SOURCE_URI="https://github.com/Paradoxianer/ScriptureGuide/archive/v$portVersion.tar.gz"
```

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	scriptureguide$secondaryArchSuffix = $portVersion
+	app:ScriptureGuide = $portVersion
+	app:ScriptureGuideManager = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	icu$secondaryArchSuffix
```
Comment (korli) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: A package name shouldn't be used, only the libs lib:*

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	cmd:awk
+	cmd:unzip
+	cmd:wget
+	lib:libsword_1.8.1$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	icu${secondaryArchSuffix}_devel
```
Comment (korli) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: A package name shouldn't be used, only the libs devel:*

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	make $jobArgs
+	cd ../ScriptureGuideManager
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	cp -R App $appsDir
+	mv $appsDir/App $appsDir/ScriptureGuide
```
Comment (korli) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: you can bring these two lines together

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	app:ScriptureGuideManager = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:awk
+	cmd:unzip
+	cmd:wget
+	lib:libsword_1.8.1$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libicui18n$secondaryArchSuffix
```
Comment (extrowerk) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: Probably you forgot to remove this

**Comment by Paradoxianer** [comment]
This should solve all problems :)

**Comment by extrowerk** [comment]
You should put docs in $docsDir, data to $dataDir.
Licenses should go in licenses, not in $appsDir. You probably don't want to have the INSTALL.htm in the package.

**Comment by Paradoxianer** [comment]
> You should put docs in $docsDir, data to $dataDir.
> Licenses should go in licenses, not in $appsDir.
Good points... but i would suggest since i need to restructure the whole project to achive this..in a good way
i would leave this enhancements to 0.9.2
(which i am working on already in the development branch)
> You probably don't want to have the INSTALL.htm in the package.
hmm it need to be reworked anyway.. good point and then it can go into the documentation forders... under a nother name :-) like for devs or so :).
but again would put this into 0.9.2.
Since scriptureguide hpkg on haikuporter is now broken i really would like to see this not perfekt but working hpkg

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe (hunk 4 above)
Comment (korli) [reply haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: this is not correct, should be cp -R App  $appsDir/ScriptureGuide

**Comment by Paradoxianer** [comment]
Thats the best i can do :)

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 6]
```diff
@@ -0,0 +1,62 @@
+SUMMARY="Bible study tool that supports a wide variety of Bibles"
+DESCRIPTION="ScriptureGuide is a Bible study tool. It supports all of the \
+features you'd expect: search, setting the font, taking notes, etc. Through \
+the accompanying ScriptureGuideManager app, you can download hundereds of free \
+copies of various Bibles, commentaries, and related books."
+HOMEPAGE="https://github.com/HaikuArchives/ScriptureGuide"
+COPYRIGHT="2019 ScriptureGuide Team"
+LICENSE="GNU GPL v2"
```
Comment (Begasus) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: There's also the MIT licence included
```diff
-LICENSE="GNU GPL v2"
+LICENSE="GNU GPL v2
+	MIT"

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 7]
```diff
@@ -0,0 +1,62 @@
+SUMMARY="Bible study tool that supports a wide variety of Bibles"
+DESCRIPTION="ScriptureGuide is a Bible study tool. It supports all of the \
+features you'd expect: search, setting the font, taking notes, etc. Through \
+the accompanying ScriptureGuideManager app, you can download hundereds of free \
+copies of various Bibles, commentaries, and related books."
+HOMEPAGE="https://github.com/HaikuArchives/ScriptureGuide"
+COPYRIGHT="2019 ScriptureGuide Team"
```
Comment (Begasus) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: Suggestion (from the license file)
```diff
-COPYRIGHT="2019 ScriptureGuide Team"
+COPYRIGHT="2004 Jan Bungeroth
+	2005-2019 ScriptureGuide Team"
```

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 8]
```diff
@@ -0,0 +1,62 @@
+SUMMARY="Bible study tool that supports a wide variety of Bibles"
+DESCRIPTION="ScriptureGuide is a Bible study tool. It supports all of the \
+features you'd expect: search, setting the font, taking notes, etc. Through \
+the accompanying ScriptureGuideManager app, you can download hundereds of free \
+copies of various Bibles, commentaries, and related books."
```
Comment (Begasus) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: Suggestion (looks better in HaikuDepot app) :)
```diff
-features you'd expect: search, setting the font, taking notes, etc. Through \
-the accompanying ScriptureGuideManager app, you can download hundereds of free \
-copies of various Bibles, commentaries, and related books."
+features you'd expect: search, setting the font, taking notes, etc. 
+Through the accompanying ScriptureGuideManager app, you can download \
+hundreds of free copies of various Bibles, commentaries, and related books."
```

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+	mkdir -p $appsDir $docDir
+	mv  App/INSTALL.htm $docDir
+	mv  App/README.htm $docDir
+	mv  App/docs/*  $docDir/
+	rm -rf App/notes
+	rm -rf App/docs
+
+	cp -R  App $appsDir/ScriptureGuide
+	addAppDeskbarSymlink $appsDir/ScriptureGuide/ScriptureGuide
+	addAppDeskbarSymlink $appsDir/ScriptureGuide/ScriptureGuideManager
```
Comment (Begasus) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: I you take in the changes for COPYRIGHT and LICENSE you could change this into:
```diff
-	cp -R  App $appsDir/ScriptureGuide
-	addAppDeskbarSymlink $appsDir/ScriptureGuide/ScriptureGuide
-	addAppDeskbarSymlink $appsDir/ScriptureGuide/ScriptureGuideManager
+	cp -R  App/ScriptureGuide* $appsDir
+	addAppDeskbarSymlink $appsDir/ScriptureGuide
+	addAppDeskbarSymlink $appsDir/ScriptureGuideManager
```

File: haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+hundreds of free copies of various Bibles, commentaries, and related books."
+HOMEPAGE="https://github.com/HaikuArchives/ScriptureGuide"
+COPYRIGHT="2004 Jan Bungeroth
+	2005-2019 ScriptureGuide Team"
+LICENSE="GNU GPL v2
+	MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/Paradoxianer/ScriptureGuide/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="d45a9be201e829a6c6c8e0cece33297899d89061f77c08e8f0a0c1a68a23c3c4"
+SOURCE_DIR="ScriptureGuide-$portVersion"
```
Comment (Begasus) [inline haiku-apps/scriptureguide/scriptureguide-0.9.1.recipe]: Whoops, empty line between SOURCE_DIR and PROVIDES :)

**Comment by Begasus** [comment]
The empty line and we're good to merge, thanks for the update (PS, maybe remove the older recipe?)

**Comment by Begasus** [comment]
@Paradoxianer Build fails on x86_64, could you look into this?
https://build.haiku-os.org/buildmaster/master/x86_64/logviewer.html?buildruns/4070/builds/15161.log

**Pull request merged**